### PR TITLE
Add version validation tests

### PR DESCRIPTION
**Pull request by pmorie**
Made version validation compliant with the spec; had to do a little bit of fussing with the client. I'm not positive all the changes are needed, will test out later and cut a client PR.

### CODE REVIEW

File: pkg/rest/apisurface.go [hunk 1]
```diff
@@ -171,15 +171,15 @@ func unpackDeprovisionRequest(r *http.Request) (*osb.DeprovisionRequest, error)
...
 	version := getBrokerAPIVersionFromRequest(r)
 	if err := s.BusinessLogic.ValidateBrokerAPIVersion(version); err != nil {
-		writeError(w, err)
+		writeError(w, err, http.StatusPreconditionFailed)
 		return
 	}
 
 	request, err := unpackLastOperationRequest(r)
 	if err != nil {
 		// TODO: This should return a 400 in this case as it is either
```
Comment (LiliC) [inline pkg/rest/apisurface.go]: Guess we can remove these comments now? :)

File: pkg/rest/apisurface.go [hunk 2]
```diff
@@ -189,7 +189,7 @@ func (s *APISurface) LastOperationHandler(w http.ResponseWriter, r *http.Request
...
```
Comment (LiliC) [inline pkg/rest/apisurface.go]: Same as above, feel free to remove these. :)

File: pkg/server/server_test.go [hunk 3]
```diff
@@ -12,16 +13,19 @@ import (
 	osb "github.com/pmorie/go-open-service-broker-client/v2"
 )
 
+// TODO: is this more of an integration test?
```
Comment (LiliC) [inline pkg/server/server_test.go]: Yes. :)

File: pkg/server/server_test.go [hunk 4]
```diff
@@ -51,47 +55,97 @@ func (b *FakeBusinessLogic) Unbind(request *osb.UnbindRequest, w http.ResponseWr
...
 
 func (b *FakeBusinessLogic) Update(request *osb.UpdateInstanceRequest, w http.ResponseWriter, r *http.Request) (*osb.UpdateInstanceResponse, error) {
 	return b.update(request, w, r)
 }
 
+func defaultValidateFunc(_ string) error {
+	return nil
+}
+
+func strPtr(s string) *string {
```
Comment (LiliC) [inline pkg/server/server_test.go]: 👍

File: pkg/server/server_test.go [hunk 5]
```diff
@@ -51,47 +55,97 @@ func (b *FakeBusinessLogic) Unbind(request *osb.UnbindRequest, w http.ResponseWr
...
+			client, err := osb.NewClient(config)
+			if err != nil {
+				t.Error(err)
+			}
+
+			actualResponse, err := client.GetCatalog()
+			if err != nil {
+				if tc.err != nil {
+					if e, a := tc.err, err; !reflect.DeepEqual(e, a) {
+						t.Errorf("unexpected error; expected %v, got %v", e, a)
```
Comment (LiliC) [inline pkg/server/server_test.go]: Nitpick: Should be uppercase:
`Unexpected error...`

File: pkg/rest/apisurface.go [hunk 6]
```diff
@@ -171,15 +171,15 @@ func unpackDeprovisionRequest(r *http.Request) (*osb.DeprovisionRequest, error)
 func (s *APISurface) LastOperationHandler(w http.ResponseWriter, r *http.Request) {
 	version := getBrokerAPIVersionFromRequest(r)
 	if err := s.BusinessLogic.ValidateBrokerAPIVersion(version); err != nil {
-		writeError(w, err)
+		writeError(w, err, http.StatusPreconditionFailed)
 		return
 	}
 
 	request, err := unpackLastOperationRequest(r)
```
Comment (LiliC) [inline pkg/rest/apisurface.go]: Oops figured out I forgot to call `unmarshalRequestBody` in the `unpackLastOperationRequest`, can create a PR for that, or if you want to change it here. Up to you?

Just figured out that there would be no errors ever returned here, which made me think why :woman_facepalming:.

Comment (pmorie) [reply pkg/rest/apisurface.go]: Let's do that one in a follow-up

File: pkg/rest/apisurface.go (hunk 1 above)
Comment (pmorie) [reply pkg/rest/apisurface.go]: Let's keep them in - they still return the wrong code, I just carried out the mechanical refactor. Let's go through and add integration tests and correct them as we add tests.

File: pkg/server/server_test.go (hunk 5 above)
Comment (pmorie) [reply pkg/server/server_test.go]: Fixed

File: pkg/rest/apisurface.go (hunk 6 above)
Comment (pmorie) [reply pkg/rest/apisurface.go]: Last Operation doesn't actually have a request body to deserialize - all the information is carried in query parameters, so we're good from that angle.

**Pull request merged**